### PR TITLE
fix: Added sub folders for protocol directory

### PR DESCRIPTION
**Pull request by yahu1031**
_Replace this paragraph with a description of what this PR is changing or adding, and why._

Listed issue is #330 

## Pre-launch Checklist

- [x] I read the [Contribute](https://docs.serverpod.dev/contribute) page and followed the process outlined there for submitting PRs.
- [x] I read and followed the [Dart Style Guide](https://dart.dev/guides/language/effective-dart/style) and formatted the code with [dart format](https://dart.dev/tools/dart-format).
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`), and made sure that the document follows the same style as other Serverpod documentation. I checked spelling and grammar.
- [ ] I added new tests to check the change I am making.
- [x] All existing and new tests are passing.
- [x] Any breaking changes are documented below.

## Breaking Changes
- Not a breaking change but a feature enhancement.

### CODE REVIEW

**Comment by yahu1031** [comment]
image

Sample output

**Review by AndryHTC — APPROVED** [review]
I had a look: the code is clean and just follows what is needed in a recursive folder search.

Not sure what happens in isolated edge cases, but I think most of the people that don't use folders inside protocols will never encounter the issue, and all the people that want the feature will be OK with this in the meantime 👍

File: tools/serverpod_cli/bin/generator/class_analyzer.dart [hunk 1]
```diff
@@ -19,23 +19,31 @@ List<ProtocolFileDefinition> performAnalyzeClasses({
...
       if (verbose) print('  - skipping file: ${entity.path}');
       continue;
     }
+    var otherFolder = entity.path
+        .replaceAll(config.protocolSourcePath, '')
+        .split(Platform.pathSeparator);
+    String? subFolder;
+    if (otherFolder.length > 2) {
+      subFolder = p.joinAll(otherFolder.sublist(0, otherFolder.length - 1));
+      if (verbose) print('Other folder: $subFolder');
```
Comment (vlidholt) [inline tools/serverpod_cli/bin/generator/class_analyzer.dart:36]: `print(' - subDirectory: ...)` to follow same standard as other printouts.

File: tools/serverpod_cli/bin/generator/class_analyzer.dart [hunk 2]
```diff
@@ -19,23 +19,31 @@ List<ProtocolFileDefinition> performAnalyzeClasses({
...
 
   for (var entity in sourceFileList) {
     if (entity is! File || !entity.path.endsWith('.yaml')) {
       if (verbose) print('  - skipping file: ${entity.path}');
       continue;
     }
+    var otherFolder = entity.path
+        .replaceAll(config.protocolSourcePath, '')
+        .split(Platform.pathSeparator);
+    String? subFolder;
```
Comment (vlidholt) [inline tools/serverpod_cli/bin/generator/class_analyzer.dart:33]: Rename `subFolder` to `subDirectory` for consistency

**Review by vlidholt — CHANGES_REQUESTED** [review]
Will be approved with the proposed minor changes.

Thanks for the contribution! :)

File: tools/serverpod_cli/bin/generator/class_analyzer.dart (hunk 1 above)
Comment (yahu1031) [reply tools/serverpod_cli/bin/generator/class_analyzer.dart]: please cross-check if this is what you mean.
```diff
- if (verbose) print('Other folder: $subFolder');
+ print('- subDirectory: $subDirectory');
```

**Pull request merged**